rts/motoko-rts/src/persistence.rs

```diff
@@ -206,6 +206,22 @@ pub unsafe fn register_stable_type<M: Memory>(
     (*metadata).stable_type.assign(mem, &new_type);
 }
 
+/// Register the stable actor type on canister initialization and upgrade.
+/// The type is stored in the persistent metadata memory for later retrieval on canister upgrades.
+/// The `new_type` value points to a blob encoding the new stable actor type.
+#[ic_mem_fn]
+pub unsafe fn assign_stable_type<M: Memory>(
```

Comment:
I wonder whether it could make sense to split `register_stable_type` in two parts, checking compatibility and assigning/storing the new stable type. The logic of `assign_stable_type` is like a subset of `register_stable_type`. Just to reduce some code duplication. Not a big deal, just an idea.

Reply:
I refactor both to use a common helper instead.